web_ui/prometheus.go

```diff
@@ -470,6 +470,12 @@ func ConfigureEmbeddedPrometheus(ctx context.Context, engine *gin.Engine) error
 	}
 	cfg.tsdb.RetentionDuration = retention
 
+	maxBytes, err := units.ParseBase2Bytes(param.Monitoring_DataRetentionSize.GetString())
+	if err != nil {
+		return fmt.Errorf("failed to parse max bytes for retention size: %v", err)
```

Comment:
I think this should explicitly point at the config knob so the user knows what they need to change.